tsdb/db.go

```diff
@@ -282,22 +293,40 @@ func newDBMetrics(db *DB, r prometheus.Registerer) *dbMetrics {
 		Name: "prometheus_tsdb_reloads_failures_total",
 		Help: "Number of times the database failed to reloadBlocks block data from disk.",
 	})
-	m.compactionsTriggered = prometheus.NewCounter(prometheus.CounterOpts{
-		Name: "prometheus_tsdb_compactions_triggered_total",
-		Help: "Total number of triggered compactions for the partition.",
```

Comment:
Here we need to keep the metric you just deleted, but we'll use `NewCounterVec` instead. For example:

```go
	m.compactionsTriggered = *prometheus.NewCounterVec(prometheus.CounterOpts{
		Name:        "prometheus_tsdb_head_compactions_triggered_total",
		Help:        "Total number of triggered compactions for the partition.",
	}, []string{"type"})
```